libraries/kmp-iap/library/src/iosMain/kotlin/io/github/hyochan/kmpiap/InAppPurchaseIOS.kt

```
@@ -1260,6 +1268,33 @@ internal class InAppPurchaseIOS : KmpInAppPurchase {
+                    if (type == ProductType.Subs.rawValue) {
+                        convertAnyListToProductSubscriptions(listOf(item)).firstOrNull()?.let {
+                            ProductOrSubscription.ProductSubscriptionItem(it)
+                        }
+                    } else {
+                        convertAnyListToProducts(listOf(item)).firstOrNull()?.let {
+                            ProductOrSubscription.ProductItem(it)
+                        }
+                    }
+                }
+            }
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)